ui/src/main/java/edu/wpi/grip/ui/codeGeneration/Exporter.java

```diff
@@ -0,0 +1,122 @@
+package edu.wpi.grip.ui.codegeneration;
+
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import org.apache.velocity.Template;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.Velocity;
+
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+
+import edu.wpi.grip.core.Pipeline;
+import edu.wpi.grip.core.Step;
+import edu.wpi.grip.core.sockets.InputSocket;
+import edu.wpi.grip.core.sockets.OutputSocket;
+import edu.wpi.grip.ui.codegeneration.java.TPipeline;
+
+@Singleton
+public class Exporter {
+  static{
+    Velocity.init();
+  }
+
+  public String stepNames(Pipeline pipeline) {
```

Comment:
This breaks the law of demeter.
The method takes a pipeline but really only needs a list of steps.